### PR TITLE
Email credentials

### PR DESCRIPTION
**Pull request by XaosSintez**

### CODE REVIEW

File: src/Services/EmailCredentialService.php [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+     * @param string $credentialId
+     * @param array|ArrayObject $params
+     *
+     * @throws NotFoundException The resource data does exist
+     *
+     * @return EmailCredential
+     */
+    public function load($credentialId, $params = [])
+    {
+        return $this->client()->get('email-credentials/{credentialId}', ['credentialId' => $credentialId] + (array) $params);
```
Comment (mdomba) [inline src/Services/EmailCredentialService.php]: could it happen that $params has the 'credentialId' parameter ?

If yes then array_merge should be used.

Comment (slavcodev) [inline src/Services/EmailCredentialService.php]: I believe it shouldn't. First argument `$credentialId` is mandatory and important, and shouldn't be overridden.

Comment (mdomba) [inline src/Services/EmailCredentialService.php]: that's why array_merge() would be better safer as it does override the array element if already exists... + does only a union

Comment (slavcodev) [inline src/Services/EmailCredentialService.php]: I'm not sure I understand you. Union is ok here, while the `credentialId` exists in first array, value from second will be ignored.

Comment (mdomba) [inline src/Services/EmailCredentialService.php]: yes, all fine... I mixed the parameter order for those 2 options... (don't worry sometimes I too don't understand myself :stuck_out_tongue_winking_eye: )

Comment (slavcodev) [inline src/Services/EmailCredentialService.php]: :+1:

**Comment by adamaltman** [comment]
A conflict needs to be fixed so that we can merge.

**Pull request merged**